Dockerfile

```diff
@@ -167,8 +167,8 @@ FROM api-runtime-private as saas-api
 RUN --mount=type=secret,id=sse_pgp_pkey \
   apt-get update && apt-get install -y gnupg && \
   gpg --import /run/secrets/sse_pgp_pkey && \
-  mv /root/.gnupg /app/; \
-  chown -R nobody /app/.gnupg
+  mv /root/.gnupg/ /app/ && \
```

Comment:
Are you able to build this image without any GPG key?

Reply:
> Are you able to build this image without any GPG key?

There's no need as `saas-api` target is only intended for building with the key.

Reply:
> > Are you able to build this image without any GPG key?
> 
> There's no need as `saas-api` target is only intended for building with the key.

Hmm, I am little confused… Is this file not used for private cloud? What if we want to build an image locally to test something?

Reply:
This file is used for all of our targets now. To build the `private-cloud` target, you don't need the key; the GPG import is the only layer added by the `saas-api` target.

I'd love to make the key a runtime dependency and shave off one target, though! Let me know if you have an idea how to do that.

See Dockerfile comments and the [PR description](https://github.com/Flagsmith/flagsmith/pull/4165#issue-2353549270) for details.